### PR TITLE
[Ingest Manager] Guard against empty stream.datasource and namespace

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This PR fallbacks to default values if namespace or datasource is an empty string

## Why is it important?

Empty string is not a valid value

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go [hunk 1]
```diff
@@ -80,6 +87,14 @@ datasources:
         - paths: /var/log/mysql/access.log
           dataset: dsds
           index: mytype-dsds-nsns
+  - name: All specified with empty strings
+    namespace: ""
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go:91]: Shouldn't the expected now be `default`?

Comment (michalpristas) [reply x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go:91]: this is test for injecting index, it does not change existing values. 
we either generate index based on these values or we're adding `add_fields` processor with correct values

Comment (ruflin) [reply x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go:91]: we need both. values and index must match. 

In a follow up we can discuss, if we should even error if it is "".

File: x-pack/elastic-agent/CHANGELOG.asciidoc [hunk 2]
```diff
@@ -39,7 +39,8 @@
 - Remove fleet admin from setup script {pull}18611[18611]
 - Correctly report platform and family. {issue}18665[18665]
 - Clean action store after enrolling to new configuration {pull}18656[18656]
- [Ingest Manager] Avoid watching monitor logs {pull}18723[18723]
+- Avoid watching monitor logs {pull}18723[18723]
```
[the remark this answers is not in the document]
Comment (michalpristas) [reply x-pack/elastic-agent/CHANGELOG.asciidoc:42]: no i'm just fixing format of previous one

File: x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go (hunk 1 above)
Comment (ph) [reply x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go:91]: @michalpristas I also thought the result should be default?

Comment (michalpristas) [reply x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go:91]: i think we can fix it but outside of injectIndex rule, what this should do is injecting index and not modifying anything on top of that.

**Pull request merged**